cartographer/mapping/internal/optimization/optimization_problem_2d.cc

```diff
@@ -301,6 +350,64 @@ void OptimizationProblem2D::Solve(
     }
   }
 
+  // Add fixed frame pose constraints.
+  std::map<int, CeresPose> C_fixed_frames;
```

Comment:
Using a `CeresPose` which is 3D here is inelegant. The rotation part is not used in the optimization below.